File: safari-600.7-branch/LayoutTests/http/tests/misc/resources/referrer-main-resource.php
<html>
<head>
<script>
function runTest() {

    if (window.testRunner)
        testRunner.dumpAsText();

    <?php if (isset($_SERVER['HTTP_REFERER']))
              echo 'document.write("FAIL: The server should not receive a referrer which is not set by user agent.");';
          else
              echo 'document.write("PASS: The server didn\'t receive a referrer.");';
    ?>;

}
</script>
</head>
<body onload="runTest()">
</body>
</html>

File: LayoutTests/http/tests/misc/resources/referrer-main-resource.py
#!/usr/bin/env python3

import os
import sys

sys.stdout.write(
    'Content-Type: text/html\r\n\r\n'
    '<html>\n'
    '<head>\n'
    '<script>\n'
    'function runTest() {\n\n'
    '    if (window.testRunner)\n'
    '        testRunner.dumpAsText();\n\n'
)

if os.environ.get('HTTP_REFERER'):
    sys.stdout.write('document.write("FAIL: The server should not receive a referrer which is not set by user agent.");')
else:
    sys.stdout.write('document.write("PASS: The server didn\'t receive a referrer.");')

sys.stdout.write(
    '}\n'
    '</script>\n'
    '</head>\n'
    '<body onload="runTest()">\n'
    '</body>\n'
    '</html>\n'
)